eng/packages/http-client-csharp-mgmt/emitter/src/resource-detection.ts

```diff
@@ -190,6 +194,18 @@ export async function updateClients(
     }
   }
 
+  // Build the unified ARM provider schema
+  const armProviderSchema = buildArmProviderSchema(
+    resourceModels,
+    resourceModelToMetadataMap,
+    nonResourceMethods,
+    sdkContext
+  );
+
+  // Apply the unified decorator to the root client
+  applyArmProviderSchemaDecorator(codeModel, armProviderSchema);
+
+  // Also apply old decorators for backward compatibility during transition
```

Comment:
we want to do the transition in this PR as well.
could you do it by removing all the old decorators and switch the processing logic in C# side as well?